packages/cli/src/lib/usecases/generate-documentation.usecase.ts

```diff
@@ -32,7 +82,23 @@ export class GenerateDocumentationUseCase {
         this.logger.info('Starting documentation generation...');
         this.logger.info(`Source directory: ${input.source}`);
         this.logger.info(`Output directory: ${input.output}`);
-        this.logger.info(`CI/CD type: ${input.type}`);
+        
+        // Log platform options if provided
+        if (input.repository?.platform) {
+            this.logger.info(`Repository platform: ${input.repository.platform}`);
+        }
+        if (input.cicd?.platform) {
+            this.logger.info(`CI/CD platform: ${input.cicd.platform}`);
+        }
+        
+        // Log section options if provided
+        if (input.sections?.includeSections?.length) {
+            this.logger.info(`Including sections: ${input.sections.includeSections.join(', ')}`);
+        }
+        if (input.sections?.excludeSections?.length) {
+            this.logger.info(`Excluding sections: ${input.sections.excludeSections.join(', ')}`);
+        }
+        
```

Comment:
If no repository platform input is given, detect it with repository service.

Same idea for cicd platform.

Then pass both to the generator service so it will be not in charge to initialize them itself